src/res_printer.ml

```diff
@@ -1423,16 +1423,21 @@ and printTypExpr (typExpr : Parsetree.core_type) cmtTbl =
         doc
     in
     Doc.concat [typ; Doc.text " as "; Doc.concat [Doc.text "'"; printIdentLike alias]]
-  | Ptyp_constr({txt = Longident.Ldot(Longident.Lident "Js", "t")}, [{ptyp_desc = Ptyp_object (_fields, _openFlag)} as typ]) ->
-    let bsObject = printTypExpr typ cmtTbl in
-    begin match typExpr.ptyp_attributes with
-    | [] -> bsObject
-    | attrs ->
-      Doc.concat [
-        printAttributes ~inline:true attrs cmtTbl;
-        printTypExpr typ cmtTbl;
-      ]
-    end
+
+  (* object printings *)
+  | Ptyp_object (fields, openFlag) ->
+    printBsObjectSugar ~inline:false fields openFlag cmtTbl
```

Comment:
Should we improve the name of this function?

Reply:
Was thinking of unifying this with record printing, which we can discuss later. I'll do this now.